src/core/api.md

```diff
@@ -129,6 +129,11 @@ export interface JSXNode<T = any> {
 // @public
 export function notifyRender(hostElement: Element): Promise<void>;
 
+// @public
+export interface Observer {
+    <T extends {}>(obj: T): T;

```

Comment:
it would be interesting to make this more strongly typed, can we observe any kind of objects? I have an idea of how to achieve this, we can develop on a call.

anyway, good to merge for now!

Reply:
Yes, let's get a discussion going.